event/src/main/java/com/bazaarvoice/emodb/event/db/astyanax/DefaultSlabAllocator.java

```diff
@@ -71,8 +73,23 @@ private static ScheduledExecutorService defaultExecutor(LifeCycleRegistry lifeCy
         return executor;
     }
 
+    // Compute how many slots in a slab will be used to do a default (new slab) allocation
+    private static int defaultAllocationCount(PeekingIterator<Integer> eventSizes) {
+        int count = 0;
+        int sum = 0;
+        while (eventSizes.hasNext()) {
+            if (sum + eventSizes.peek() <= Constants.MAX_SLAB_SIZE_IN_BYTES && count + 1 <= Constants.MAX_SLAB_SIZE) {
```

Comment:
It may be more efficient to flip the conditions in this conditional. Because we know that the bulk of our traffic is small messages, it is more likely that the we will reach 1000 events before we run out of space in the slab.